apps/bare-expo/android/app/build.gradle

```diff
@@ -219,3 +219,45 @@ task copyDownloadableDepsToLibs(type: Copy) {
     from configurations.compile
     into 'libs'
 }
+
+/**
+ * Download a precompiled version of React Native
+ */
+buildscript {
+  repositories {
+    jcenter()
+  }
+  dependencies {
+    classpath 'de.undercouch:gradle-download-task:4.0.0'
+  }
+}
+apply plugin: "de.undercouch.download"
+def downloadsDir = new File(buildDir, "downloads")
+
+afterEvaluate {
+  downloadsDir.mkdirs()
+  // Fetch from NPM
+  def packageUrl = "https://registry.npmjs.org/react-native/-/react-native-${reactNativeVersion}.tgz"
```

Comment:
We want to use the copy from react-native-lab. bare-expo will compile it here: https://github.com/expo/expo/blob/master/apps/bare-expo/scripts/setup-android-project.sh#L15-L31

Reply:
1. Do you know if the commit https://github.com/expo/react-native/commit/2a2663ff532439d7b8a98a1179558eb9e7824224 is breaking in any way (`facebook/react-native` vs `expo/react-native`)?
2. If so, maybe we should execute that script as part of the project setup (so that RN is always resolved)? Sometimes we would like to use `bare-expo` to test our changes in unimodules in bare workflow, so we would open it with Android Studio. 🤔

Reply:
That commit probably breaks things. Reading it, it needs to be cleaned up so it is something we can sustainably keep in the future. Probably `handleReloadJS` should detect whether the Expo classes exist, and run the logic for `reloadExpoApp`. The goal is to allow RN to work in both a bare app and in the Expo client. Opened an issue here: https://github.com/expo/expo/issues/6262

The script I linked to runs when you run the Yarn scripts in bare-expo. We could add it to the bare-expo Gradle file so that it always runs when you try to build the Android app.